admin_panel/exams/serializer.py

```diff
@@ -64,6 +67,16 @@ class Meta:
             'total_marks',
             'organization',
             'created_at',
-            'updated_at'
+            'updated_at',
+            'batch_count',
+            'student_count'
         ]
         read_only_fields = ['id', 'created_at', 'updated_at']
+
+    def get_batch_count(self, obj):
+        return obj.exam_batches.count()
+
+    def get_student_count(self, obj):
+        return obj.exam_batches.aggregate(
+            total_students=models.Count('exam_students')
+        )['total_students'] or 0
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider optimizing student count query</b></div>
<div id="fix">

Consider using `annotate()` instead of `aggregate()` for better query optimization when getting student count. The current implementation may cause N+1 query issues when serializing multiple exam objects.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -79,4 +79,4 @@
  def get_student_count(self, obj):
 -        return obj.exam_batches.aggregate(
 -            total_students=models.Count('exam_students')
 -        )['total_students'] or 0
 +        return getattr(obj, 'student_count', obj.exam_batches.aggregate(total_students=models.Count('exam_students'))['total_students'] or 0)
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/11#issuecomment-2518952072>#96a4d8</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged